app/services/deploy_service.rb

```diff
@@ -21,16 +21,17 @@ def start
   private
 
   def create_github_pull_request
-    pull_request = github_client.create_pull_request(
-      github_repo,
-      deploy_strategy.arguments['base'],
-      deploy_strategy.arguments['head'],
-      'Deploy',
-      'This is an automatically generated release PR!'
-    )
-
-    assignee = choose_assignee(pull_request)
-    github_client.add_assignees(github_repo, pull_request.number, assignee) if assignee
+    if can_release_now?(deploy_strategy.arguments['blocked_time_buckets'] || [])
+      pull_request = github_client.create_pull_request(
+        github_repo,
+        deploy_strategy.arguments['base'],
+        deploy_strategy.arguments['head'],
+        'Deploy',
+        'This is an automatically generated release PR!'
+      )
+      assignee = choose_assignee(pull_request)
+      github_client.add_assignees(github_repo, pull_request.number, assignee) if assignee
+    end
```

Comment:
I think this works, but I'm noticing that the logic has become complex. There are all of these code locations testing whether the deploy can proceed:

https://github.com/artsy/horizon/blob/54d9646244b1ef9aae60184fa4cb607ed71a3742/app/services/comparison_service.rb#L64

And (here):

```
if can_release_now?(deploy_strategy.arguments['blocked_time_buckets'] || [])
```

And: 

https://github.com/artsy/horizon/blob/54d9646244b1ef9aae60184fa4cb607ed71a3742/app/services/deploy_service.rb#L41-L46

In a later PR, maybe we should refactor these into a single condition that prevents the whole `DeployService...` from being invoked.

Reply:
Indeed, I think I will be able to do a refactor near in future once I get more familiar with horizon.